spec/lib/audit/moab_to_catalog_spec.rb

```diff
@@ -35,62 +35,59 @@
   end
 
   describe '.seed_catalog_for_all_storage_roots' do
-    it 'calls seed_catalog_for_dir with the right argument once per root' do
-      expect(described_class).to receive(:seed_catalog_for_dir).exactly(MoabStorageRoot.count).times
+    before do
+      allow(described_class).to receive(:seed_catalog_for_dir).exactly(MoabStorageRoot.count).times
       MoabStorageRoot.pluck(:storage_location) do |path|
-        expect(described_class).to receive(:seed_catalog_for_dir).with("#{path}/#{Settings.moab.storage_trunk}")
+        allow(described_class).to receive(:seed_catalog_for_dir).with("#{path}/#{Settings.moab.storage_trunk}")
       end
+    end
+
+    it 'calls seed_catalog_for_dir with the right argument once per root' do
       described_class.seed_catalog_for_all_storage_roots
+      expect(described_class).to have_received(:seed_catalog_for_dir).exactly(MoabStorageRoot.count).times
+      MoabStorageRoot.pluck(:storage_location) do |path|
+        expect(described_class).to have_received(:seed_catalog_for_dir).with("#{path}/#{Settings.moab.storage_trunk}")
+      end
```

Comment:
I apologize for muddling this PR by changing this spec to use `have_recieved` - but it helped me work on the failures to be addressed and it wasnt a very long file.

The changes highlighted here are solely for refactoring to use `have_recieved`